### ISSUE TITLE
Fix aspect-ratio of custom background images

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-android#466]
custom background images are always shown as "scale-to-screen" which does not respect the aspect ration.

i would suggest to show background images always with the correct aspect ration and cut parts top/bottom or left/right (so that the image is centered in the screen rectangle).

https://github.com/deltachat/deltachat-android/issues/460#issuecomment-439395579 shows some images

there is some code in the old telegram-ui that handles this by on-the-fly-creating a temporary bitmap cropped from the orignal one iirc.

**Comment by florianhaar** [comment]
I will look if I can fix this :-)

**Comment by r10s** [comment]
cool :)

**Comment by florianhaar** [comment]
@r10s do you have an idea where i can find the old telegram code for this issue?

**Comment by r10s** [comment]
@florianhaar in general, the telegram code is at https://github.com/deltachat/deltachat-android-old , i think greping for "background" or so should give some further details. please let me know if you get stuck, i can have a closer look then.

**Comment by r10s** [comment]
as the issue tracker should show only things actually being working on i close this for now.

**Comment by angelo-fuchs** [comment]
Related: If one changes the background image to a different image of same name, the image is not recreated and no change is visible.

**Comment by r10s** [comment]
@angelo-fuchs reopened that and reverted the pr as this led to "jumping" wallpapers whenever eg. the soft-keyboard is opened or closed.

**Comment by r10s** [comment]
to summarize the last attempts:

- #1372 fixes the issue by using a dedicated ImageView for the background and applying centerCrop to it. this, however, opens the issue that the background image is no longer "still", it jumps up and down when the view is resized (eg. when the softkeyboard opens)

- #1424 fixes the issue by croppig the image to the correct size, one for landscape, one for portrait.  
however, as this is done only once after image selection, screen size changes (split screen ...) would probably not work this way. also, selected background images were aborted, see the pr itself for some more details, it needed. also, the need to maintain two images may cause problems in the future.

parts of #1424, however, are already merged into master: the on-config-change-update and the landscape-default image.

for finally fixing the issue, i think, we should
- use an ImageView as in #1372 (i found it complicated to get the size of the activity and to make some sense of that)
- however, more important, when setting the image, crop it to the correct size. directly before display, this way, no two images are needed. we could use glide for that and let glide cache the image (we disabled most glide caches as they caused problems, however, in this explicit case it makes probably sense)
- we could move all the logic to a `BackgroundImageView` class

**Comment by angelo-fuchs** [comment]
@r10s I did notice the movement of the background but personally preferred it. So, you sure that the old state is better?
Not that it could not be improved upon but for people who have a custom background the "jumping" solution is notably better then the "stretching" state we had before.
I would have preferred to leave the "jumping" situation and open an issue on it, not happy to see it reverted.

I will see that I find some time to work on this in the next days.

**Comment by r10s** [comment]
> @r10s I did notice the movement of the background but personally preferred it.

:)

it is not so much noticable with the default background - but this is also true for the bad aspect-ratio.

> So, you sure that the old state is better?

definitively :) changing most of the visible screen content, shifting things around differently, is nothing a user would expect when expanding the soft-keyboard. compared to that, the wrong aspect-ratio is no really big thing - esp. assuming that the set background is more an impression, some artwork etc. - in most cases you cannot even see it completely.

also if you have a look at whatsapp/telegram/whatgram etc. - the background is stable there, nothing shifts around.

> I will see that I find some time to work on this in the next days.

thanks, i also prepared some things, did not finish that to a pr yet, however.

but i think, it is only a tiny bit left, at least, we have a pretty good idea what we should do now.

btw. if we apply the same default to the default image, we can drop the landscape-variants, that were introduced to have a similiar image impression, and just use a _square_ - this way, we save some 100k of space and have some images fewer to maintain.
